Source/WebCore/loader/cache/CachedFontLoadRequest.h

```diff
@@ -63,35 +64,37 @@ class CachedFontLoadRequest final : public FontLoadRequest, public CachedFontCli
     bool ensureCustomFontData() final { return m_font->ensureCustomFontData(); }
     RefPtr<Font> createFont(const FontDescription& description, bool syntheticBold, bool syntheticItalic, const FontCreationContext& fontCreationContext) final
     {
-        return m_font->createFont(description, syntheticBold, syntheticItalic, fontCreationContext);
+        return protectedCachedFont()->createFont(description, syntheticBold, syntheticItalic, fontCreationContext);
     }
 
     void setClient(FontLoadRequestClient* client) final
     {
-        auto* oldClient = m_fontLoadRequestClient;
+        WeakPtr oldClient = m_fontLoadRequestClient;
         m_fontLoadRequestClient = client;
 
         if (!client && oldClient)
-            m_font->removeClient(*this);
+            protectedCachedFont()->removeClient(*this);
         else if (client && !oldClient)
-            m_font->addClient(*this);
+            protectedCachedFont()->addClient(*this);
```

Comment:
I wonder if we really need all this implementation code in a header.

Reply:
I doubt it. Would you live me to introduce a cpp in this PR and move the implementation there? There is no cpp currently

Reply:
No! But might be a nice follow-up (as part of our "fewer include parse workload" project)

Reply:
That way we can revert just that one change if it turns out to be a performance impact.